File: section0-9/flattenRecursive.py
def flatten(arr):
    if len(arr)==0: return []
    finalArray = []
    if type(arr[0]) is list:
        finalArray = flatten(arr[0])
    else:
        finalArray.append(arr[0])
    return finalArray+flatten(arr[1:])

print(flatten([1, 2, 3, [4, 5] ]))
print(flatten([1, [2, [3, 4], [[5]]]]))
print(flatten([[1],[2],[3]]))
print(flatten([[[[1], [[[2]]], [[[[[[[3]]]]]]]]]]))

File: section0-9/flattenRecursive.js
function flatten(arr) {
    if (arr.length === 0) return [];
    let array = [];
    Array.isArray(arr[0]) ? array = flatten(arr[0]) : array.push(arr[0])
    return array.concat(flatten(arr.slice(1)));
}

console.log(flatten([1, [2, [3, 4], [[5]]]]))